### ISSUE TITLE
col_schema: less strict matching of types and no type matching

### ISSUE THREAD

**Issue by tomryder9** [rstudio/pointblank#124]
I really love the col_schema_match() function and its helper function col_schema(). It's helping to ensure that the columns and types for a transformed table are exactly what we think they should be.

There are some possible enhancements for col_schema(), all in the name of making this less strict (good in some circumstances):

1. Have an option to use all_of() and any_of() for the column types. Columns like datetime have POSIXct and POSIXt and it would be nice to just specify one of those with any_of("POSIXct")

2. Have it so we don't need to specify the column types (if we just want to ensure the column names are present)

Thanks again for your awesome work in this really awesome package! Like I said before, it is such an incredible contribution.

**Comment by rich-iannone** [comment]
Hello @tomryder9 , I worked on your feature request and I’m quite pleased in how it turned out. I took your two-part suggestion and was able to satisfy the requirements by adding an argument (`is_exact`). Have a look at the updated documentation for all the details, I think you’ll like the implementation.

Thanks for posting this issue. It took some time to get to it, but I’m glad I did. Feature requests like this strengthen the project. I was able to add an extra set of tests and make corrections I might not otherwise have noticed if it weren’t for this issue!
